### PR TITLE
e2e tests initial commit

### PR DESCRIPTION
**Pull request by pweil-**
@simon3z @enoodle this introduces a set of files that we can use for writing e2e tests.  Most of these are straight from the openshift repo but I've removed a lot of things that aren't relevant.

Example of this running:

image

### CODE REVIEW

File: test/end-to-end/e2e.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
```
Comment (pweil-) [inline test/end-to-end/e2e.sh]: @simon3z @enoodle this is the actual test(s) that are running

**Comment by simon3z** [comment]
@pweil- can you mention somewhere where the libs were taken from? So maybe we'll be able to update them in the future from the same source. Thanks!

File: test/end-to-end/e2e.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${OS_ROOT}/hack/util.sh"
+source "${OS_ROOT}/hack/cmd_util.sh"
+os::log::install_errexit
+
+source "${OS_ROOT}/hack/lib/util/environment.sh"
+os::util::environment::setup_time_vars
+
+# test args
+os::cmd::expect_failure_and_text "image-inspector --help" "Usage of"
```
Comment (moolitayer) [inline test/end-to-end/e2e.sh]: I wonder why we do not return 0 for --help?

**Comment by pweil-** [comment]
> can you mention somewhere where the libs were taken from? So maybe we'll be able to update them in the future from the same source

updated scripts with `# see <link>`

**Comment by simon3z** [comment]
> updated scripts with # see <link>

@pweil- for some some reason I am not seein it.

@enoodle @moolitayer please continue reviewing. Thanks.

**Comment by pweil-** [comment]
> @pweil- for some some reason I am not seeing it.

heh, forgot my `--force` tag.  It was just sitting there with an error

File: test/end-to-end/e2e.sh (hunk 2 above)
Comment (enoodle) [inline test/end-to-end/e2e.sh]: I think its a default from the library that parses the flags.

**Comment by simon3z** [comment]
I appreciate the direction here but I can't review this all. Although if we're taking/rebasing these out of origin there's probably not too much to discuss (we should take it up to the openshift repo).

@enoodle please give it another look and then we'll merge this.

**Comment by enoodle** [comment]
I run it and its working for me well. The tests looks clear and its easy enough to understand what they do.
@pweil- My only question is how to use the functions from ```hack/util.sh```? I didn't see them being used in any other place but they seem useful for the next tests that we will want to make (testing the serve option and the openscap scan)

**Comment by pweil-** [comment]
@enoodle The functions from `util.sh` are sourced in to `e2e.sh` and are being used to run the tests you see there.  Right now only two of them are being used in that file but I suspect we'll use more of them as we add more tests.  

There are lots more examples of things like timed urls, and try until in https://github.com/openshift/origin/blob/master/test/end-to-end/core.sh.

**Comment by pweil-** [comment]
Oh, I see what you're asking.  I was referencing cmd_util and not just util.  Not all of those are used but since they looked useful I left them in :smile:   We can look for examples in the origin codebase for usages or remove anything that isn't useful.  I didn't want to delete too much right away.

**Comment by simon3z** [comment]
:+1: for me @enoodle ready for merge?

**Comment by enoodle** [comment]
LGTM :+1: 
Those functions looks like things we will want to learn to use very soon.

**Pull request merged**